--- generated-sources/sablecc/de/deutschdiachrondigital/dddquery/node/Node.java	2009-09-01 15:14:56.000000000 +0200
+++ dddqueryNode.java	2009-09-01 15:09:55.000000000 +0200
@@ -63,13 +63,13 @@
         return null;
     }
 
-    protected <T> List<T> cloneList(List<T> list)
+    protected <T extends Node> List<T> cloneList(List<T> list)
     {
         List<T> clone = new LinkedList<T>();
 
         for(T n : list)
         {
-            clone.add(n);
+            clone.add(cloneNode(n));
         }
 
         return clone;
