--- src/sablecc/de/deutschdiachrondigital/dddquery/node/Node.java	2008-03-31 15:15:50.000000000 +0200
+++ src//sablecc-patched/de/deutschdiachrondigital/dddquery/node/Node.java	2008-03-31 15:18:54.000000000 +0200
@@ -63,13 +63,13 @@
         return null;
     }
 
-    protected <T> List<T> cloneList(List<T> list)
+    protected <T extends Node> List<T> cloneList(List<T> list)
     {
         List<T> clone = new LinkedList<T>();
 
         for(T n : list)
         {
-            clone.add(n);
+            clone.add(cloneNode(n));
         }
 
         return clone;
